javascripts/discourse/connectors/topic-list-item/item.gjs

```diff
@@ -0,0 +1,174 @@
+import Component from "@glimmer/component";
+import { get, hash } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import ShareTopicModal from "discourse/components/modal/share-topic";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import TopicExcerpt from "discourse/components/topic-list/topic-excerpt";
+import TopicLink from "discourse/components/topic-list/topic-link";
+import UnreadIndicator from "discourse/components/topic-list/unread-indicator";
+import TopicPostBadges from "discourse/components/topic-post-badges";
+import TopicStatus from "discourse/components/topic-status";
+import categoryLink from "discourse/helpers/category-link";
+import icon from "discourse/helpers/d-icon";
+import discourseTags from "discourse/helpers/discourse-tags";
+import formatDate from "discourse/helpers/format-date";
+import topicFeaturedLink from "discourse/helpers/topic-featured-link";
+import { wantsNewWindow } from "discourse/lib/intercept-click";
+import DiscourseURL from "discourse/lib/url";
+import { i18n } from "discourse-i18n";
+
+export default class Item extends Component {
+  @service currentUser;
+  @service historyStore;
+  @service modal;
+
+  get newDotText() {
+    return this.currentUser?.trust_level > 0
+      ? ""
+      : i18n("filters.new.lower_title");
+  }
+
+  @action
+  onTitleFocus(event) {
+    event.target.closest(".topic-list-item").classList.add("selected");
+  }
+
+  @action
+  onTitleBlur(event) {
+    event.target.closest(".topic-list-item").classList.remove("selected");
+  }
+
+  @action
+  openTopic(event) {
+    if (
+      (event.target.nodeName === "A" && !event.target.closest(".raw-link")) ||
+      event.target.closest(".badge-wrapper")
+    ) {
+      return;
+    }
+
+    const { topic } = this.args.outletArgs;
+
+    if (wantsNewWindow(event)) {
+      window.open(topic.lastUnreadUrl, "_blank");
+    } else {
+      this.historyStore.set("lastTopicIdViewed", topic.id);
```

Comment:
Having this core implementation detail in a theme isn't ideal... I wonder if we should pass a `visitTopic` action down into the plugin outlet?